src/Network/Network/PrivateLinkService/PrivateLinkServiceProvider/PrivateLinkProviderFactory.cs

```diff
@@ -0,0 +1,50 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.Network.PrivateLinkService.PrivateLinkServiceProvider
+{
+    public class PrivateLinkProviderFactory
+    {
+        private const string NETWORKING_TYPE = "microsoft.network/privatelinkservices";
+        
+        public static IPrivateLinkProvider CreatePrivateLinkProvder(NetworkBaseCmdlet cmdlet, string subscription, string privateLinkResourceType)
+        {
+            IPrivateLinkProvider provider = null;
+
+            if(privateLinkResourceType == null)
+            {
+                return new NetworkingProvider(cmdlet);
+            }
+
+            if(GenericProvider.SupportsPrivateLinkResourceType(privateLinkResourceType))
+            {
+                return new GenericProvider(cmdlet, subscription, privateLinkResourceType);
+            }
+
+            switch (privateLinkResourceType.ToLower())
+            {
+                case NETWORKING_TYPE:
+                default:
```

Comment:
for unknown type, should we return error or throw exceptions?

Reply:
It's network original logic. If resource type is unknown, returns NetworkProvider.